helix-term/src/application.rs

```diff
@@ -910,6 +795,24 @@ impl Application {
                         diagnostics.sort_unstable_by_key(|(d, server_id)| {
                             (d.severity, d.range.start, *server_id)
                         });
+
+                        if let Some(doc) = doc {
+                            doc.replace_diagnostics(
+                                Editor::doc_diagnostics_with_filter(
+                                    &self.editor.language_servers,
+                                    &self.editor.diagnostics,
+                                    doc,
+                                    |diagnostic, ls_id| {
+                                        ls_id == server_id
+                                            && diagnostic.source.as_ref().map_or(true, |source| {
+                                                !unchanged_diag_sources.contains(source)
+                                            })
+                                    },
+                                ),
```

Comment:
nit: make a seperate variable for this. It's a bit unreadable with so much nesting

Reply:
I've made it slightly less nested in the latest changes...